zkevm-circuits/src/evm_circuit/execution/extcodehash.rs

```diff
@@ -263,4 +263,28 @@ mod test {
             test_ok(Some(account), false);
         }
     }
+
+    #[test]
+    // Regression test to ensure that the code hash for an account that is is being initialized is
+    // the empty code hash.
+    fn create_tx_extcodehash() {
```

Comment:
I think I'm missing something while reading this test.  How is this test checking that the initialized account has the empty code hash?

Reply:
The tx does not have `to` set, so it is an account creation tx. The result ADDRESS and then EXTCODEHASH should be the code hash of the current account, which in this case will be keccak(nil). Previously in https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/1079/commits/b682805243f4f7c73571af56511830b8c0bf9885, this would produce an bus-mapping error, as we would claim that the code hash was 0 instead.

Reply:
Got it, thanks a lot for the explanation!